docs/egress-ip.md

```diff
@@ -13,45 +14,97 @@ For more info, consider looking at the following links:
+- The rule with `100` priority is added for the pod matching `egressip-prod` EgressIP, and it redirects the traffic to one of the egress nodes (ECMP is used to balance the traffic between next hops).
 
+Once the redirected traffic reaches one of the egress nodes it gets SNATed in the gateway router:
+```shell
+ovn-nbctl lr-nat-list GR_ovn-worker
+TYPE             GATEWAY_PORT          EXTERNAL_IP        EXTERNAL_PORT    LOGICAL_IP          EXTERNAL_MAC         LOGICAL_PORT
+...
+snat                                   172.18.0.33                         10.244.1.3
+ovn-nbctl lr-nat-list  GR_ovn-worker2
+TYPE             GATEWAY_PORT          EXTERNAL_IP        EXTERNAL_PORT    LOGICAL_IP          EXTERNAL_MAC         LOGICAL_PORT
+...
+snat                                   172.18.0.44                         10.244.1.3
+```
+
+### Pod to node IP traffic
+When a cluster networked pod matched by an egress IP tries to connect to a non-local node IP it hits the following
+logical router policy in `ovn_cluster_router`:
+```shell
+# $<all_eip_pod_ips> - address-set of all pod IPs matched by any EgressIP
+# $<all_esvc_pod_ips> - address-set of all pod IPs matched by any EgressService
+# $<all_node_ips> - address-set of all node IPs in the cluster
+102 (ip4.src == $<all_eip_pod_ips> || ip4.src == $<all_esvc_pod_ips>) && ip4.dst == $<all_node_ips>  allow    pkt_mark=1008
+```
+In addition to simply allowing the `pod -> node IP` traffic so that EgressIP reroute policies 
+are not matched upon, it is also marked with the 1008 mark.  
+If a pod is hosted on an egressNode the traffic will first get SNATed to the egress IP, and then it will hit 
+following flow on breth0 that will SNAT the traffic to local node IP:
+```shell
+# output truncated, 0x3f0 == 1008
+priority=105,pkt_mark=0x3f0,ip,in_port=2 actions=ct(commit,zone=64000,nat(src=<NodeIP>),exec(load:0x1->NXM_NX_CT_MARK[])),output:1
+``` 
+This is required to make `pod -> node IP` traffic behave the same regardless of where the pod is hosted.  
+Implementation details: https://github.com/ovn-org/ovn-kubernetes/commit/e2c981a42a28e6213d9daf3b4489c18dc2b84b19.
+
+### Dealing with non SNATed traffic
+Egress IP is often configured on a node different from the one hosting the affected pods.  
+Due to the fact that ovn-controllers on different nodes apply the changes independently,
+there is a chance that some pod traffic will reach the egress node before it configures the SNAT rules.
+The following flows are added on breth0 to address this scenario:
+```shell
+# Commit connections from local pods so they are not affected by the drop rule below, this is required for ICNIv2
+priority=109,ip,in_port=2,nw_src=<nodeSubnet> actions=ct(commit,zone=64000,exec(set_field:0x1->ct_mark)),output:1
+
+# Drop non SNATed egress traffic coming from non-local pods
+priority=104,ip,in_port=2,nw_src=<clusterSubnet> actions=drop
+
+# Commit connections coming from IPs not in cluster network
```

Comment:
self-note: not really related to this feature but good to have in docs. Its a more wider capture all from patch-port default flow but yes as far as egressIP traffic is concerned it will either hit 104 (srcIP=cluster-cidr-excluding-local-node's-pod-cdir) OR 100 (srcIP=EIP) on a node on its way out.